### PR TITLE
Add support for async collection methods

### PR DESCRIPTION
**Pull request by jdgjsag67251**
See #288.

Ok, so apparently Meteor just internally calls the non-async methods and wraps them in a Promise: https://github.com/meteor/meteor/blob/devel/packages/mongo/collection.js#L998. 
This PR just adds tests to make sure that the hooks keep working when Meteor changes their implementation.

### CODE REVIEW

File: collection-hooks.js [hunk 1]
```diff
@@ -66,6 +64,16 @@ CollectionHooks.extendCollectionInstance = function extendCollectionInstance (se
           }
         }
       }
+    }
+
+    Object.entries(advices).forEach(function ([method, advice]) {
```
Comment (Jantje19) [inline collection-hooks.js]: I'm an idiot. I totally did this in the wrong place.

Comment (jdgjsag67251) [inline collection-hooks.js]: I am an idiot. I totally did this in the wrong place.

File: tests/async.js [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+  Tinytest.addAsync('async - before - findOneAsync', async (test, next) => {
+    const collection = new Mongo.Collection(null)
+
+    let called = false
+
+    collection.before.findOne(() => {
+      called = true
+    })
+
+    const id = collection.insert({ test: true })
```
Comment (StorytellerCZ) [inline tests/async.js:30]: Use `insertAsync` as above

**Review by StorytellerCZ — CHANGES_REQUESTED** [review]
Please use *Async methods when calling DB.
Once we merge this we should first move to support async functions in hooks and then move the hooks internals to use the new async functions to be ready for Meteor 3.

**Comment by jdgjsag67251** [comment]
Cool, thnx. I've updated the PR

**Review by StorytellerCZ — APPROVED** [review]

**Review by jankapunkt — APPROVED** [review]

**Comment by jankapunkt** [comment]
That's great @jdgjsag67251 :+1:  Thanks a lot!

**Pull request merged**